### ISSUE TITLE
Handling MIDI input

### ISSUE THREAD

**Issue by carlo-bramini** [Mindwerks/wildmidi#208]
Hello,
I would like to send to WildMidi some MIDI events received by a physical instruments connected to a MIDI port or via USB.
But it seems to me that the library does not provide a function for sending directly MIDI events to the generator. None of the functions exported by the DLL seem to be good for this purpose.
At the moment I implemented a code for opening a MIDI input port and receive the MIDI events on Windows, and this was the easy part.
Now I would like to send to the library those events, like note_on for example, and here I'm blocked.
Have you some suggestions for this problem?

Sincerely.

**Comment by Tchibo9000** [comment]
Hi!
Sorry, post is two years old, but has still no answer.
As far as I understand WildMIDI is a simple software midi player. It should include a core software synthesizer (softsynth) library that can be use with other applications. But actually I really have not found it in the source tree.

**Comment by carlo-bramini** [comment]
Hello,
I did some changes to the library and I implemented the function for sending the events to the synthesizer.
But I remember that it has not been so easy...
Perhaps, this issue could be used as a reminder for adding a new feature in the future.

**Comment by Tchibo9000** [comment]
Absolutely! A clear MIDI interface in wildmidi for the synthesizer would be more valuable than just to have a sequencer.

I did something similar for CSharpSynthProject a few years ago and remember it being difficult too.
